util/types/src/core/cell.rs

```diff
@@ -15,8 +15,12 @@ pub struct CellMeta {
     pub cell_output: CellOutput,
     pub out_point: OutPoint,
     pub transaction_info: Option<TransactionInfo>,
-    pub data_bytes: u64,
     /// In memory cell data and its hash
+    pub data_size: u64,
+    /// Data occupied capacity
+    /// may not equals to actually data size since virtual occupied
+    pub data_occupied_capacity: Capacity,
```

Comment:
I think only one field `occupied_capacity` is enough, it's `virtual_occupied` or data's size.

Reply:
Is `data_size` enough? What will happen if `data_size` does not equal to the real data size?

Reply:
IMO we should keep both fields.

* `data_size` must be equals to real data size. we use `data_size` to check the validity of a syscall before actually load the `data` [details](https://github.com/jjyr/ckb/blob/satoshi-gift/script/src/syscalls/load_cell_data.rs#L91).
* `occupied_capacity` is the fields leave to us to hack the `occupied`.

We use the distinguished fields, to break the hidden-convention of `data_size` to `occupied capacity`.

It's clear enough, what's benefit can we get for using only one "overload" "hidden-convention" field?